rholang/src/test/scala/coop/rchain/rholang/interpreter/ReduceSpec.scala

```diff
@@ -2059,4 +2056,284 @@ class ReduceSpec extends FlatSpec with Matchers with PersistentStoreTester {
+        Await.result(toSetTask.runToFuture, 3.seconds)
+    }
+    val resultList =
+      ESetBody(
+        ParSet(
+          List(
+          )
+        )
+      )
+    result.exprs should be(Seq(Expr(resultList)))
+    errorLog.readAndClearErrorVector().unsafeRunSync should be(Vector.empty[InterpreterError])
+  }
+
+  it should "keep Set(1, 2, 3) unchanged" in {
+    implicit val errorLog: ErrorLog[Task] = new ErrorLog[Task]()
+    val toSetCall: EMethod =
+      EMethod(
+        "toSet",
+        ESetBody(
+          ParSet(
+            List(
+              GInt(1L),
+              GInt(2L),
+              GInt(3L)
+            )
+          )
+        ),
+        List()
+      )
+
+    val result: Par = withTestSpace(errorLog) {
+      case TestFixture(_, reducer) =>
+        implicit val env: Env[Par] = Env[Par]()
+        val toSetTask             = reducer.evalExpr(toSetCall)
+        Await.result(toSetTask.runToFuture, 3.seconds)
+    }
+    val resultList =
+      ESetBody(
+        ParSet(
+          List(
+            GInt(1L),
+            GInt(2L),
+            GInt(3L)
+          )
+        )
+      )
+    result.exprs should be(Seq(Expr(resultList)))
+    errorLog.readAndClearErrorVector().unsafeRunSync should be(Vector.empty[InterpreterError])
+  }
+
+
+  it should "return an error when `toSet` is called with arguments" in {
+    implicit val errorLog: ErrorLog[Task] = new ErrorLog[Task]()
+    val toSetCall: EMethod =
+      EMethod(
+        "toSet",
+        ESetBody(
+          ParSet(
+            List(
+              GInt(1L),
+              GInt(2L),
+              GInt(3L)
+            )
+          )
+        ),
+        List(GInt(1))
+      )
+
+    val result = withTestSpace(errorLog) {
+      case TestFixture(space, reducer) =>
+        implicit val env = Env[Par]()
+        val nthTask      = reducer.eval(toSetCall)
+        val inspectTask  = nthTask >> space.toMap
+        Await.result(inspectTask.runToFuture, 3.seconds)
+    }
+    result should be(mutable.HashMap.empty)
+    errorLog.readAndClearErrorVector().unsafeRunSync should be(
+      Vector(MethodArgumentNumberMismatch("toSet", 0, 1))
+    )
+  }
+
+  it should "return an error when `toSet` is called on GInt" in {
+    implicit val errorLog: ErrorLog[Task] = new ErrorLog[Task]()
+    val toSetCall: EMethod =
+      EMethod(
+        "toSet",
+        GInt(1L),
+        List()
+      )
+
+    val result = withTestSpace(errorLog) {
+      case TestFixture(space, reducer) =>
+        implicit val env = Env[Par]()
+        val nthTask      = reducer.eval(toSetCall)
+        val inspectTask  = nthTask >> space.toMap
+        Await.result(inspectTask.runToFuture, 3.seconds)
+    }
+    result should be(mutable.HashMap.empty)
+    errorLog.readAndClearErrorVector().unsafeRunSync should be(
+      Vector(MethodNotDefined("toSet", "Int"))
+    )
+  }
+
+
+  "toMap" should """transform [("a",1), ("b",2), ("c",3)] into {"a":1, "b":2, "c":3)""" in {
```

Comment:
What happens if `toMap` is called on lists like `["a", ("b", 2)]` or `[1, ("b", 2)]`?

Reply:
Please add a test that shows that a list like `[("a", 1), ("a", 2)]` is transformed into a map `{ "a" : 1 }`.